dotnet/src/Connectors/VectorData.Abstractions/VectorStorage/IVectorStoreRecordCollection.cs

```diff
@@ -129,4 +131,16 @@ public interface IVectorStoreRecordCollection<TKey, TRecord> : IVectorizedSearch
     /// <exception cref="VectorStoreOperationException">The command fails to execute for any reason.</exception>
     /// <exception cref="VectorStoreRecordMappingException">The mapping between the storage model and record data model fails.</exception>
     IAsyncEnumerable<TKey> UpsertAsync(IEnumerable<TRecord> records, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Gets matching records from the vector store. Does not guarantee that the collection exists.
```

Comment:
Unrelated to this PR: I see that we have "Does not guarantee that the collection exists" on each API here - we probably don't need to specify this on each and every API... Hopefully users understand that this is just a .NET object and cannot possibly guarantee anything about what's going on in the database (FWIW we don't have such warnings in EF). At most I'd add a remark at the type level...